src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -3289,37 +3235,48 @@ private void login(String primary, String primaryInstanceName, int primaryPortNu
             // the network with requests, then update sleep interval for next iteration (max 1 second interval)
             // We have to sleep for every attempt in case of non-dbMirroring scenarios (including multisubnetfailover),
             // Whereas for dbMirroring, we sleep for every two attempts as each attempt is to a different server.
-            if (!isDBMirroring || (1 == attemptNumber % 2)) {
+            if (SQLServerException.DRIVER_ERROR_INTERMITTENT_TLS_FAILED == driverErrorCode
+                    && tlsRetryAttempt < INTERMITTENT_TLS_MAX_RETRY && !timerHasExpired(timerExpire)) {
```

Comment:
It looks like we won't enter this if block and retry intermittent TLS errors because on line 3195 there's
```
|| SQLServerException.DRIVER_ERROR_INTERMITTENT_TLS_FAILED == driverErrorCode // TLS1.2 failure
```
which if true, will rethrow the error instead of moving on to the if block.